### PR TITLE
Feature/guilds status modal

### PR DESCRIPTION
**Pull request by luchux**
This PR Does: 
- Adds useClickOutside hook to keep an event on mouse down out of an element and call a callback. 
- use useClickOutside in the menus scheme and status filters, and member menu, to close them when clicking out. 
- Refactor styles of MenuContent so we can tell if should show a menu in fullscreen in mobile. 

And implements https://github.com/DXgovernance/dxvote/issues/435 (figmas wireframes in the issue) 
- Adds modal to Join DXDAo and join button on sidebar. 
- Adds menu on sidebar for when the user is member already. 

 
 video: 
https://watch.screencastify.com/v/OhX9yX38AwJzZQ8ik9Wt

### CODE REVIEW

**Review by hhamud — DISMISSED** [review]
The Build fails 
image

File: src/configs/localhost/config.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "cache": {
     "fromBlock": 0,
-    "toBlock": 76,
-    "ipfsHash": "QmS3hfS68mgeBp5xfyRhBeJ4hdWMMbDSSVUfVBpnaQ3oEw"
+    "toBlock": 75,
+    "ipfsHash": "QmVz2vSovUfKuvtugoo8AtSJnV6cXiqixVsA8TLgv4Fxzv"
```
Comment (hhamud) [inline src/configs/localhost/config.json:5]: was this supposed to be committed and pushed here?

File: src/configs/proposalTitles.json [hunk 2]
```diff
@@ -1,66 +1,58 @@
 {
-  "0x08382f2197b647b338a15bcb33287dc3edc278f8c089a12952b87ee3fbadb42a": "Update swapr.eth resolver to point to beta 8.1 release of Swapr",
```
Comment (hhamud) [inline src/configs/proposalTitles.json:2]: Same thing? was this supposed to be changed here?

File: src/hooks/Guilds/useClickOutside.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { useEffect } from 'react';
+
+export const useClickOutside = (ref, callback) => {
```
Comment (hhamud) [inline src/hooks/Guilds/useClickOutside.ts]: you should add the comment here again on the purpose of this hook. I don't immediately know what `useClickOutside` is supposed to do. Or you can make the name clearer

**Review by hhamud — DISMISSED** [review]
It works great! but there are a few issues.

**Comment by luchux** [comment]
@hhamud we are ok with the configs, we can re generate them anytime. I put things in a lot of commits stupidly, and its hard to revert and remove to the actual state of main. 
So Ross, gave the ok to go with them as their to avoid much handling.

**Review by hhamud — APPROVED** [review]
LGTM!

File: src/components/Guilds/Sidebar/GuestActions.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+`;
+
+export const GuestActions = ({ onJoin }) => {
+  const [joinModal, setJoinModal] = useState(false);
+  const [dxdValue, setDXDValue] = useState(0);
+  return (
+    <>
+      <Button onClick={() => setJoinModal(true)}>Join</Button>
+      <Modal
+        header={<ModalHeader>Join DxDao</ModalHeader>}
```
Comment (rossneilson) [inline src/components/Guilds/Sidebar/GuestActions.tsx:99]: "DXdao" is the correct formatting here

File: src/components/Guilds/Sidebar/GuestActions.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  font-size: 20px;
+  font-weight: 600;
+`;
+
+const ButtonLock = styled(Button)`
+  width: 100%;
+  margin-top: 22px;
+`;
+
+export const GuestActions = ({ onJoin }) => {
```
Comment (rossneilson) [inline src/components/Guilds/Sidebar/GuestActions.tsx:92]: Could we make the padding more like the figma design in this component?
Currently a large space below the header and very cramped at the bottom

**Comment by luchux** [comment]
@rossneilson merged this and will update your changes and new changes that were added today at the figma in another issue ticket.

**Pull request merged**